Code/Framework/AzCore/AzCore/UnitTest/UnitTest.h

```diff
@@ -181,9 +182,37 @@ namespace UnitTest
         }
     };
 
+    class TraceBusHook;
+
+    // Helper class to automatically re-enable the TraceBusHook when going out of scope
+    // This is necessary because the TraceBusHook is part of the Environment which does not reset after each test run
+    // So we need to ensure the bus is always reconnected between tests
+    struct TraceBusHookUniqueDisableToken
+    {
+        TraceBusHookUniqueDisableToken(TraceBusHook* traceBusHook);
+        ~TraceBusHookUniqueDisableToken();
+
+        void ReEnable();
+
+    private:
+        TraceBusHook* m_traceBusHook{};
+    };
```

Comment:
There is already a way to disable the TraceBusHook.
That is to not use the [AZ_UNIT_TEST_HOOK](https://github.com/o3de/o3de/blob/development/Code/Framework/AzTest/AzTest/AzTest.h#L447)

Reply:
Looks like that applies to the entire module?  We need something more fine grained so specific tests can control how suppression works